script/check_newsfragments.py

```diff
@@ -0,0 +1,68 @@
+# Licensed under the GPL: https://www.gnu.org/licenses/old-licenses/gpl-2.0.html
+# For details: https://github.com/PyCQA/pylint/blob/main/LICENSE
+# Copyright (c) https://github.com/PyCQA/pylint/blob/main/CONTRIBUTORS.txt
+
+"""Small script to check the formatting of news fragments for towncrier.
+Used by pre-commit.
+"""
+
+from __future__ import annotations
+
+import argparse
+import re
+import sys
+from pathlib import Path
+from re import Pattern
+
+VALID_ISSUES_KEYWORDS = [
+    "Refs",
+    "Closes",
+    "Follow-up in",
+    "Fixes part of",
+]
+ISSUES_KEYWORDS = "|".join(VALID_ISSUES_KEYWORDS)
+VALID_CHANGELOG_PATTERN = rf"(?P<description>(.*\n)*(.*\.\n))\n(?P<ref>({ISSUES_KEYWORDS}) (PyCQA/astroid)?#(?P<issue>\d+))"
+VALID_CHANGELOG_COMPILED_PATTERN: Pattern[str] = re.compile(
+    VALID_CHANGELOG_PATTERN, flags=re.MULTILINE
+)
+
+
+def main(argv: list[str] | None = None) -> int:
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "filenames",
+        nargs="*",
+        metavar="FILES",
+        help="File names to check",
+    )
+    parser.add_argument("--verbose", "-v", action="count", default=0)
+    args = parser.parse_args(argv)
+    verbose = args.verbose
+    is_valid = True
+    for filename in args.filenames:
+        file = Path(filename)
+        is_valid &= check_file(file, verbose)
+    return 0 if is_valid else 1
```

Comment:
`sys.exit` works with booleans so we can just use booleans although `bool(True)` is `1`. So we would need to slightly modify to `check_if_incorrect` or something.

Reply:
I'm not quite sure if I understand what you mean with "slightly modify to `check_if_incorrect` or something"? But the hint with `sys.exit` is nice, that makes it simpler.

Reply:
I meant that currently we return `is_valid` but that evaluates to `True` which is `1` which gives `sys.exit(1)` so we'll fail when `is_valid` is `True`. So we need to turn the conditional around to have `check_file` return `is_invalid`.

Reply:
Aaah, got you. In that case I'd rather leave it as it is right now. Imo the negation of the return value would be similar confusing as the current ternary if, whereas the latter has the benefit that it's a bit clearer we want to return the actual exit code of the program.

Reply:
Wouldn't switching `True` and `False` around and renaming to `check_missing_entry` already work? Oh well, never mind!